src/test/rpc/AccountObjects_test.cpp

```diff
@@ -316,10 +316,225 @@ class AccountObjects_test : public beast::unit_test::suite
         }
     }
 
+    void testObjectTypes()
+    {
+        testcase ("object types");
+
+        // Give gw a bunch of ledger objects and make sure we can retrieve
+        // them by type.
+        using namespace jtx;
+
+        Account const alice { "alice" };
+        Account const gw{ "gateway" };
+        auto const USD = gw["USD"];
+
+        // Test for ticket account objects when they are supported.
+        Env env(*this, supported_amendments().set(featureTickets));
+
+        // Make a lambda we can use to get "account_objects" easily.
+        auto acct_objs = [&env] (Account const& acct, char const* type)
+        {
+            Json::Value params;
+            params[jss::account] = acct.human();
+            params[jss::type] = type;
+            params[jss::ledger_index] = "validated";
+            return env.rpc("json", "account_objects", to_string(params));
+        };
+
+        // Make a lambda that easily identifies empty account objects.
+        auto acct_objs_is_empty =
+            [&env, &acct_objs] (Account const& acct, char const* type)
+        {
+            Json::Value const resp = acct_objs (acct, type);
+
+            bool const isArray = resp["result"]["account_objects"].isArray();
+            return isArray && (resp["result"]["account_objects"].size() == 0);
+        };
+
+        env.fund(XRP(10000), gw, alice);
+        env.close();
+
+        // Since the account is empty now, all account objects should come
+        // back null or empty.
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::account));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::amendments));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::check));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::directory));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::escrow));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::fee));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::hashes));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::offer));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::payment_channel));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::signer_list));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::state));
+        BEAST_EXPECT (acct_objs_is_empty (gw, jss::ticket));
+
+        // Set up a trust line so we can find it.
+        env.trust(USD(1000), alice);
+        env.close();
+        env(pay(gw, alice, USD(5)));
+        env.close();
+        {
+            // Find the trustline and make sure it's the right one.
+            Json::Value const resp = acct_objs (gw, jss::state);
+
+            BEAST_EXPECT (resp["result"]["account_objects"].isArray());
```

Comment:
Consider using the defined json fields instead.
eg. `resp[jss::result][jss::account_objects]`